### PR TITLE
Benefit from the static metadata for GraalVM to have been moved into …

### PR DESCRIPTION
**Pull request by Sanne**
…Hibernate ORM upstream

### CODE REVIEW

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmReflections.java [hunk 1]
```diff
@@ -1,161 +0,0 @@
...
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.BooleanLiteralNode.class);
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.SimpleCaseNode.class);
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.AbstractMapComponentNode.class);
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.FromReferenceNode.class);
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.OrderByClause.class);
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.FromClause.class);
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.ConstructorNode.class);
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.LiteralNode.class);
-        simpleConstructor(reflectiveClass, org.hibernate.hql.internal.ast.tree.BinaryArithmeticOperatorNode.class);
-        simpleConstructor(reflectiveClass, CommonToken.class);
```
Comment (gwenneg) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmReflections.java]: There's a native test failure that seems related to that line. Was it moved upstream too?

Comment (Sanne) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmReflections.java]: thanks, looks like something is wrong.  The upstream metadata is rather different, I've been trying to optimise it further - I did run all Quarkus tests during its development, but looks like I screwed something up during the last polishing phase. I'll re-introduce this file so to add the missing entry.

**Comment by Sanne** [comment]
I've re-pushed to fix the problem.

File: extensions/hibernate-orm/runtime/pom.xml [hunk 2]
```diff
@@ -36,6 +36,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-narayana-jta</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-graalvm</artifactId>
+        </dependency>
```
Comment (gwenneg) [inline extensions/hibernate-orm/runtime/pom.xml]: This needs to be removed since it's a duplicate of the other `hibernate-graalvm` dependency.

**Review by gwenneg — CHANGES_REQUESTED** [review]

File: extensions/hibernate-orm/runtime/pom.xml (hunk 2 above)
Comment (Sanne) [reply extensions/hibernate-orm/runtime/pom.xml]: thanks for spotting that! Done

**Review by gwenneg — APPROVED** [review]

**Comment by machi1990** [comment]
There is a test failure, I am unsure if it is related 

```
2020-02-18T13:21:42.5881201Z [INFO] [INFO] ------------------------------------------------------------------------
2020-02-18T13:21:42.5899996Z [INFO] [INFO] Total time:  7.142 s
2020-02-18T13:21:42.5906586Z [INFO] [INFO] Finished at: 2020-02-18T13:21:42Z
2020-02-18T13:21:42.5912249Z [INFO] [INFO] ------------------------------------------------------------------------
2020-02-18T13:21:42.6101317Z [INFO] run post-build script verify.groovy
2020-02-18T13:21:42.6114163Z [INFO] Running post-build script: /home/vsts/work/1/s/integration-tests/spring-data-jpa/target/it/spring-configuration/verify.groovy
2020-02-18T13:21:43.5444020Z [INFO]           spring-configuration/pom.xml ..................... FAILED (9.5 s)
2020-02-18T13:21:43.5493890Z [INFO]   The post-build script did not succeed. assert contentFile.contains("Quarkus does not support the following Spring Boot configuration properties: ")
2020-02-18T13:21:43.5494953Z        |           |
2020-02-18T13:21:43.5495615Z        |           false
2020-02-18T13:21:43.5599254Z        '[INFO] Scanning for projects...\n[INFO] \n[INFO] -------------------< org.acme:spring-configuration >--------------------\n[INFO] Building spring-configuration 1.0-SNAPSHOT\n[INFO] --------------------------------[ jar ]---------------------------------\n[INFO] \n[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ spring-configuration ---\n[INFO] Using \'UTF-8\' encoding to copy filtered resources.\n[INFO] Copying 1 resource\n[INFO] \n[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ spring-configuration ---\n[INFO] Changes detected - recompiling the module!\n[INFO] Compiling 1 source file to /home/vsts/work/1/s/integration-tests/spring-data-jpa/target/it/spring-configuration/target/classes\n[INFO] \n[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ spring-configuration ---\n[INFO] Using \'UTF-8\' encoding to copy filtered resources.\n[INFO] skip non existing resourceDirectory /home/vsts/work/1/s/integration-tests/spring-data-jpa/target/it/spring-configuration/src/test/resources\n[INFO] \n[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ spring-configuration ---\n[INFO] No sources to compile\n[INFO] \n[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ spring-configuration ---\n[INFO] No tests to run.\n[INFO] \n[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ spring-configuration ---\n[INFO] Building jar: /home/vsts/work/1/s/integration-tests/spring-data-jpa/target/it/spring-configuration/target/spring-configuration-1.0-SNAPSHOT.jar\n[INFO] \n[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:build (default) @ spring-configuration ---\n[INFO] [org.jboss.threads] JBoss Threads version 3.0.1.Final\n[INFO] [org.hibernate.Version] HHH000412: Hibernate ORM core version 5.4.12.Final\n[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building thin jar: /home/vsts/work/1/s/integration-tests/spring-data-jpa/target/it/spring-configuration/target/spring-configuration-1.0-SNAPSHOT-runner.jar\n[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 2578ms\n[INFO] ------------------------------------------------------------------------\n[INFO] BUILD SUCCESS\n[INFO] ------------------------------------------------------------------------\n[INFO] Total time:  7.142 s\n[INFO] Finished at: 2020-02-18T13:21:42Z\n[INFO] ------------------------------------------------------------------------\nRunning post-build script:
```

/cc @Sanne

**Comment by Sanne** [comment]
thanks @machi1990 ! I don't think those failures are related: rebased, also to trigger another build.

**Review by gwenneg — APPROVED** [review]

**Pull request merged**